app.go

```diff
@@ -54,13 +57,20 @@ type App interface {
 }
 
 var app App
+var appLock sync.RWMutex
```

Comment:
Pity we don't have more benchmarks so we can see the performance impact of all this locking

Reply:
Agreed.
I was about to suggest that is write-once somehow - but other issues want to run multiple apps one after the other in the same runtime so I don't think that's an option. Unless we say "no" to running app2 after app1

Reply:
Hmm I think that would break our tests as we need multiple apps to run after each other

Reply:
oh yes :)